src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectOptionsProcessor.cs

```diff
@@ -31,6 +34,23 @@ public VisualStudioProjectOptionsProcessor(VisualStudioProject project, HostWork
 
             // Set up _commandLineArgumentsForCommandLine to a default. No lock taken since we're in the constructor so nothing can race.
             ReparseCommandLine_NoLock();
+
+            _optionService = workspaceServices.GetRequiredService<IOptionService>();
+
+            // For C#, we need to listen to the options for NRT analysis 
+            // that can change in VS through tools > options
+            if (_project.Language == LanguageNames.CSharp)
+            {
+                _optionService.OptionChanged += OptionService_OptionChanged;
+            }
+        }
+
+        private void OptionService_OptionChanged(object sender, OptionChangedEventArgs e)
+        {
+            if (e.Option.Feature == FeatureOnOffOptions.UseNullableReferenceTypeAnalysis.Feature)
```

Comment:
You'll want to include in name too; I'd be a bit worried that otherwise a change to any other feature on/off is retriggering a bunch of work unnecessarily.